### PR TITLE
Concatenated strings to template strings

### PR DESCRIPTION
**Pull request by greggman**
#15506

### CODE REVIEW

**Comment by mrdoob** [comment]
I was not aware you were talking about the examples folder. With this change a bunch of examples will stop working in [oldish browsers](https://caniuse.com/#feat=template-literals).

File: examples/js/nodes/core/FunctionNode.js [hunk 1]
```diff
@@ -6,7 +6,7 @@
 import { TempNode } from './TempNode.js';
 import { NodeLib } from './NodeLib.js';
 
-var declarationRegexp = /^([a-z_0-9]+)\s([a-z_0-9]+)\s*\((.*?)\)/i,
+var declarationRegexp = /^(?:(?:\s*\/\/.*?\n)*)\s*([a-z_0-9]+)\s([a-z_0-9]+)\s*\((.*?)\)/i,
```
Comment (mrdoob) [inline examples/js/nodes/core/FunctionNode.js]: Do you mind explaining this regexp change? Is it to handle tabs and comments?

Comment (yushijinhun) [reply examples/js/nodes/core/FunctionNode.js]: > Do you mind explaining this regexp change? Is it to handle tabs and comments?

This ignores line comments before function declaration.

Comment (yushijinhun) [reply examples/js/nodes/core/FunctionNode.js]: Without this,
```js
new FunctionNode( /* glsl */`
	// https://stackoverflow.com/questions/36174431/how-to-make-a-wave-warp-effect-in-shader
	vec2 sineWave(vec2 uv, vec2 phase) {
		// wave distortion
		float x = sin( 25.0*uv.y + 30.0*uv.x + 6.28*phase.x) * 0.01;
		float y = sin( 25.0*uv.y + 30.0*uv.x + 6.28*phase.y) * 0.03;
		return vec2(uv.x+x, uv.y+y);
	}
`);
```
would be broken.

**Pull request closed without merging**